### PR TITLE
Chapter 13 Dialogs

### PR DESCRIPTION
**Pull request by alexandragnj**
## New features

- Added a dialog in which user can change the date of a crime

## Changes in code

- Creating a DialogFragment, DatePickerDialog
- Add DatePickerDialog to FragmentManager
- Pass data to DatePickerDialog by creating and setting fragment arguments
- Returning data to CrimeFragment implementing a setFragmentResultListener

## Screenshot(GIF)
May-04-2022 12-48-09

### CODE REVIEW

**Review by Nicu1991 — APPROVED** [review]

File: app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt [hunk 1]
```diff
@@ -97,13 +111,21 @@ class CrimeFragment : Fragment() {
...
             isChecked = crime.isSolved
             jumpDrawablesToCurrentState()
         }
     }
 
+    override fun onFragmentResult(requestCode: String, result: Bundle) {
+        when (requestCode) {
+            DIALOG_DATE -> {
+                Log.d(TAG, "RECEIVED RESULT FOR $requestCode")
+                crime.date = DatePickerFragment.getSelectedDate(result)
```
Comment (iugaliviu) [inline app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt:124]: This looks a bit weird. Is this the way the book recommends getting the selected date?

File: app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt [hunk 2]
```diff
@@ -3,17 +3,19 @@ package com.example.criminalintentapp
...
 import android.widget.Button
 import android.widget.CheckBox
 import android.widget.EditText
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentResultListener
 import androidx.lifecycle.ViewModelProvider
 import java.util.UUID
 
-class CrimeFragment : Fragment() {
+class CrimeFragment : Fragment(), FragmentResultListener {
```
Comment (iugaliviu) [inline app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt:18]: Just as in CrimeListFragment, I would pass the layout to the `Fragment()` constructor

**Review by iugaliviu — COMMENTED** [review]
Just a few comments and one question, otherwise looks fine. Also, I would suggest moving codelines related to views from `onCreateView` to `onViewCreated`. The viewmodel asignment can stay in `onCreateView`. Google docs mention this is the best practice as `onViewCreated` guarantees 100% that there is a view to operate on

File: app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt (hunk 1 above)
Comment (alexandragnj) [reply app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt:124]: No, this is a solution that I found on the internet for passing the date between fragments using setFragmentResultlListener(), setFragmentResult() and onFragmentResult(). The book used callbacks interface and deprecated function setTargetFragment

**Pull request merged**